lib/Onyx.ts

```diff
@@ -729,10 +729,9 @@ function update(data: OnyxUpdate[]): Promise<void> {
         }
     });
 
-    return clearPromise
-        .then(() => Promise.all(promises.map((p) => p())))
-        .then(() => updateSnapshots(data))
-        .then(() => undefined);
+    const snapshotPromises = updateSnapshots(data);
+
+    return clearPromise.then(() => Promise.all([...snapshotPromises, ...promises].map((p) => p()))).then(() => undefined);
```

Comment:
Hmm I'm not a fan of this spread since that can be slow for large updates. Can we just do a concat?

Reply:
This is a pretty dense line of code too, and it is hard to see what it is supposed to be doing. Would you mind splitting it up into a couple of different lines/variables so it's a little more readable?